git is required for visual test addon via chromatic cli

weeksling opened the pull request:
This adds an error state to the panel, disables the run button, and shows a notification if the addon is run in a project without git configured.

To test this:
 - Run storybook with addon installed.
 - Rename or delete .git directory to trigger an error.
 - It should show a single notification and update the panel with the git error.

Note: this currently stops the gitObservable loop from occurring and requires the server to be restarted after fixing the issue. Added language in the notification to prompt the user to restart storybook after fixing.

Note: This does not handle build errors once the gitInfo can be retrieved. The pr this branches off of should handle build-level errors including the git one commit issue.


<!-- GITHUB_RELEASE PR BODY: canary-version -->
<details>
  <summary>📦 Published PR as canary version: <code>0.0.99--canary.126.8f69032.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @chromaui/addon-visual-tests@0.0.99--canary.126.8f69032.0
  # or 
  yarn add @chromaui/addon-visual-tests@0.0.99--canary.126.8f69032.0
  ```
</details>
<!-- GITHUB_RELEASE PR BODY: canary-version -->

weeksling commented:
@MichaelArestad I added the notification here with the same title and subheader but added a final callout to restart the storybook server. Does that seem alright to you? 
<img width="1100" alt="Screen Shot 2023-10-03 at 7 42 34 PM" src="https://github.com/chromaui/addon-visual-tests/assets/5882088/73197133-5550-457e-ade6-65eef9beac37">

tmeasday commented on src/index.ts:
```diff
@@ -26,17 +27,23 @@ function managerEntries(entry: string[] = []) {
+      prev = gitInfo;
+      timer = setTimeout(act, interval);
+    } catch (e: any) {
+      console.error("getGitInfo failed. Ending loop.", e);
```
```suggestion
      console.error(`Failed to fetch git info, with error:\n${e}`);
```

I'm not sure the stack trace is really all that helpful.

tmeasday commented on src/index.ts:
```diff
@@ -99,9 +106,19 @@ async function serverChannel(
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const gitInfoError = useAddonState<Error>(channel, GIT_INFO_ERROR);
+
+  observeGitInfo(
+    5000,
+    (info) => {
+      gitInfoState.value = info;
+    },
+    (error: Error) => {
+      gitInfoError.value = error;
+    }
+  );
```
Theoretically, it's possible there's some other issue with git (like it gets corrupted or something). I'm not sure we really need to cope with that, but just flagging it here (a git failure doesn't necessarily mean git isn't setup).

weeksling replied:
That makes sense. In this case, whatever the error occurring would otherwise fail to start the cli, so I think we'd want to handle it similarly to how we do. I think the stacktrace will be useful if users encounter other types of errors that cause this.

weeksling replied:
Nevermind. This was another issue with my code.

tmeasday reviewed (approved):
LGTM and works well in testing.

I wonder if the notification would be weird for first time SB users who haven't even opened the addon yet. Perhaps we should play it cool and wait for them to reach the addon before showing them the message?

weeksling commented:
> I wonder if the notification would be weird for first time SB users who haven't even opened the addon yet. Perhaps we should play it cool and wait for them to reach the addon before showing them the message?

I agree. Removing the notification. I also updated it to keep the loop going so they don't have to restart storybook-server. The issue I thought I had with unsetting an addon state variable was an issue in my code

weeksling commented:
> > @MichaelArestad I added the notification here with the same title and subheader but added a final callout to restart the storybook server. Does that seem alright to you?
> 
> @weeksling In what scenario would folks see this? Is it necessary?

No I don't think this is necessary. I removed the notification entirely. Users will see the error when they try to check the addon panel. I thought this was necessary to draw attention to, but it will be happening most often for new storybooks so it's noisy.

MichaelArestad commented:
Perfect. Thank you!
